src/lib/renderPaperCheckoutLink .ts

```diff
@@ -10,78 +8,62 @@ async function sleepForSeconds(seconds: number) {
     }, seconds * 1000);
   });
 }
-const PaperCheckoutResult: CheckoutSuccessResult = { transactionId: '' };
-function getPaperCheckoutResult() {
-  return PaperCheckoutResult;
-}
 
-export async function renderPaperCheckoutLink({
+export function renderPaperCheckoutLink({
   checkoutLinkUrl,
-}: // type = 'DRAWER',
-{
+  onPaymentSucceeded,
+  onPaymentFailed,
+  onTransferSucceeded,
+  onModalClosed,
+}: {
   checkoutLinkUrl: string;
-  // type: 'MODAL' | 'DRAWER';
-}): Promise<CheckoutSuccessResult> {
-  const promiseToReturn = new Promise<CheckoutSuccessResult>(
-    async (resolve, rej) => {
-      const drawer = new Drawer();
+  onPaymentSucceeded?: (transactionId: string) => void;
+  onPaymentFailed?: (transactionId: string) => void;
+  onTransferSucceeded?: (transactionId: string, claimedTokens: any) => void;
```

Comment:
If we want to provide additional args to these callbacks in the future, it would require a breaking change to add new arguments. Can we make these args all objects?

```typescript
onPaymentSucceeded?: ({ transactionId: string }) => void;

onPaymentFailed?: ({ transactionId: string }) => void;

onTransferSucceeded?: ({
  transactionId: string;
  claimedTokens: any;
}) => void;
```